scipy/signal/filter_design.py

```diff
@@ -248,6 +248,84 @@ def freqz(b, a=1, worN=None, whole=0, plot=None):
     return w, h
 
 
+def grpdelay(b, a=1, w=None, whole=False):
+    r"""Compute the group delay of a digital filter.
+
+    The group delay shows by how many samples different spectral components of
+    a signal are delayed by a filter. It is formally defined as
+    :math:`D(\omega) = -\frac{d}{d \omega} \arg H(e^{j \omega})`, where
+    :math:`H(e^{j \omega})` is the frequency response of a filter and
+    continuous (unwrapped) phase is used for differentiation.
+
+    Parameters
+    ----------
+    b : array-like
+        Numerator of a filter transfer function.
+    a : array-like, optional
+        Denominator of a filter transfer function.
+    w : {None, int, array-like}, optional
+        If None (default), then compute at 512 frequencies equally spaced
+        around the unit circle.
+        If a single integer, then compute at that many frequencies.
+        If an array-like, compute the delay at the frequencies given (in
+        radians/sample).
+    whole : boolean, optional
+        Normally, frequencies are computed from 0 to the Nyquist frequency,
+        pi radians/sample (upper-half of unit-circle).  If `whole` is True,
+        compute frequencies from 0 to 2*pi radians/sample.
+
+    Returns
+    -------
+    w : ndarray
+        The normalized frequencies at which gd was computed, in radians/sample.
+    gd : ndarray
+        The group delay.
+
+    Notes
+    -----
+    If the transfer function :math:`H(z)` has zeros or poles on the unit
+    circle then the group delay at corresponding frequencies is undefined.
+    When such case arises the warning is raised and the group delay
+    is set to 0.
+
+    Examples
+    --------
+    >>> from scipy import signal
+    >>> b, a = signal.iirdesign(0.1, 0.3, 5, 50, ftype='cheby1')
+    >>> w, gd = signal.grpdelay(b, a)
+
+    >>> import matplotlib.pyplot as plt
+    >>> plt.title('Digital filter group delay')
+    >>> plt.plot(w, gd)
+    >>> plt.ylabel('Group delay [samples]')
+    >>> plt.xlabel('Frequency [rad/sample]')
+    >>> plt.show()
+    """
+    if w is None:
+        w = 512
+    if isinstance(w, int):
+        if whole:
+            w = numpy.linspace(0, 2 * pi, w, endpoint=False)
```

Comment:
Can you use ``np`` instead of ``numpy`` here and in the rest of the function? This file contains some other uses of ``numpy``, but that's outdated style